lib/benchee/benchmark.ex

```diff
@@ -43,9 +43,23 @@ defmodule Benchee.Benchmark do
   @spec measure(Suite.t, module) :: Suite.t
   def measure(suite = %Suite{jobs: jobs, configuration: config}, printer \\ Printer) do
     printer.configuration_information(suite)
-    run_times = record_runtimes(jobs, config, printer)
-
-    %Suite{suite | run_times: run_times}
+    results = record_results(jobs, config, printer)
+    run_times = parse_results(results, fn({key, {run_times, _}}) ->
+      {key, run_times}
+    end)
+    memory_usages = parse_results(results, fn({key, {_, memory_usages}}) ->
+      {key, memory_usages}
+    end)
```

Comment:
Just a gut feeling, but it feels like we already sort of parsing/distinguishing between run_times and memory_usages further down... would be cool if that distinction was more localized in one spot or easier to do further one. Not entirely sure how to though :)